pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -491,6 +494,143 @@ func TestReconciler(t *testing.T) {
 		wantEvents        []utiltesting.EventRecord
 		wantErr           error
 	}{
+		"PodsReady is set to False before Workload is Admitted": {
```

Comment:
Do the test cases cover 100% of the code in `generatePodsReadyCondition`? You can generate the coverage data by adding `-coverprofile=` while running unit tests, then report with `go tool cover -html`.

Reply:
It covers 95.8% of the code, the remaining ~4% is because we don't test the `default` code of block when the conditions are invalid

Reply:
Thanks for checking. 

> we don't test the default code of block when the conditions are invalid

What does it mean the conditions are invalid? Does it mean the "reason" field is unrecognized? Is this the scenario which will happen to every workload on upgrade? If so then I believe it is worth covering. 

Actually, I believe we should have some form of a test demonstrating the behavior on upgrade. We have two options: keep the old reason, but make sure it is handled as WaitForPodsStart, or convert it to "WaitForPodsStart" on the first iteration. I would be leaning to convert it actually.

Reply:
We do convert it on the first iteration, I'll cover that in tests

Reply:
Added the test, covering 100% now